### DISCUSSION TITLE
ultimateshell_v3.58.33

### DISCUSSION THREAD

**Discussion by G3G4X5X6** [G3G4X5X6/ultimate-cube#21] [Announcements]
#### :memo: Added
- 新增终端相关配置
- 实现SSH相关的会话右键菜单，如`关闭当前`等
- `最近会话` 实现点击表头排序

#### :calendar: Fixed
- 修复终端配色不生效的BUG
- 修复ssh同时保存密码登录和公钥登录时互相覆盖的bug
- 专注模式  
  - 修复专注模式退出时丢失面板的Bug
  - 修复退出专注模式时选中选项卡不正确的问题
  - 修复专注模式关闭选项卡后与退出模式时不一致的Bug
  - 修复开启专注模式时，选中SSH选项卡不一致的Bug
- 修复 `最近会话` 中打开会话时，不区分登录方式的Bug  
- 修复编辑会话时生成新的会话文件的bug

#### :card_index: Improved
- 改进最近会话生成逻辑
- 改进重新连接会话功能

#### :scissors: 使用
- 程序依赖运行环境：JDK11+
- 建议使用自带依赖的通用版本 UltimateShell-3.58.33-jar-with-dependencies.jar
- 无 JDK 环境的 Windows 用户建议使用 ultimateshell_setup.exe 安装包（体积较大）
- 关于该版本的任何疑问，请于 Discussions 中提问、反馈

## What's Changed
* Create CODE_OF_CONDUCT.md by @G3G4X5X6 in https://github.com/G3G4X5X6/ultimateshell/pull/15

**Full Changelog**: https://github.com/G3G4X5X6/ultimateshell/compare/v3.55.28...v3.58.33

<hr /><em>This discussion was created from the release <a href='https://github.com/G3G4X5X6/ultimateshell/releases/tag/v3.58.33'>ultimateshell_v3.58.33</a>.</em>

**Comment by Jonnyan404** [comment]
凉凉,直接无法打开ssh了~~~  会话管理,双击和右键打开会话,都没有反应了   😂

**Reply by G3G4X5X6** [reply]
是新版的会话文件格式和旧版的不一致，需要手动改改文件名，最近会话也清空重新生成吧（手动狗头保命）
![image](https://user-images.githubusercontent.com/87740076/156960391-f17e9818-6a86-477b-8337-a38fb90340b4.png)

**Reply by Jonnyan404** [reply]
> 是新版的会话文件格式和旧版的不一致，需要手动改改文件名，最近会话也清空重新生成吧（手动狗头保命） ![image](https://user-images.githubusercontent.com/87740076/156960391-f17e9818-6a86-477b-8337-a38fb90340b4.png)

ok,好了.  还有些小问题: 1.终端配色不能预览,调起来很麻烦. 2.外部集成功能,希望能自带模板,方便修改.

**Reply by G3G4X5X6** [reply]
1. 配色确实麻烦，有空看看怎么弄吧，这个属于花哨的功能了，有缘再见，哈哈
2. 外部集成功能,在wiki中有模板的

**Reply by Jonnyan404** [reply]
> 1. 配色确实麻烦，有空看看怎么弄吧，这个属于花哨的功能了，有缘再见，哈哈
> 2. 外部集成功能,在wiki中有模板的

好的.
还有最后一个:如果连接错误之后,面板上有个左右来回动的进度条,一直来来回回的动,看的强迫症想抠屏了...  下个版本也一起修下吧~~~

**Reply by G3G4X5X6** [reply]
这个有的，我看着也烦，所以弄了下，双击进度条，即可需取消一次连接的进度条
